src/enzyme_ad/jax/Passes/CanonicalizeFor.cpp

```diff
@@ -676,6 +678,68 @@ bool canMoveOpOutsideWhile(Operation *op, WhileOp loop) {
   return true;
 }
 
+class truncProp final : public OpRewritePattern<TruncIOp> {
```

Comment:
out of curiosity why do we need to do this kind of propagation?

Reply:
the emitted code has this =/